# 1. Background of the Study

## Title of the Study
### **Classifying Filipino Families by Income Group Using Socio-Demographic and Expenditure Data**

***

## Source of Data
| Attribute | Details |
| :--- | :--- |
| **Dataset Name** | Family Income and Expenditure Survey (FIES) — Philippines |
| **Primary Source** | Philippine Statistics Authority (PSA) |
| **Acquisition Source** | Kaggle — Family Income and Expenditure Dataset |

***

## Brief Description of the Dataset
The dataset provides detailed information on Filipino households’ **annual income**, **sources of earnings**, and comprehensive **patterns of expenditure**. It contains both quantitative (numerical) and categorical features that describe the household’s socio-economic profile.

* **Number of Rows and Columns:** 41544 rows and 60 columns
* **Sample Variables:**
    * `Region` – geographic region of the household (Categorical)
    * `Household_Head_Age` – age of the family head (Quantitative)
    * `Education` – highest educational attainment (Categorical/Ordinal)
    * `Family_Size` – total number of family members (Quantitative)
    * `Total_Income` – total annual family income (Quantitative)
    * `Food_Expenditure`, `Education_Expenditure`, `Housing_Expenditure`, etc. – detailed spending categories (Quantitative)

***

## Machine Learning Task Details

### Selected Features (Independent Variables)
The model will utilize a combination of socio-demographic and spending-related factors as input features, including:
* Region
* Education level and age of household head
* Family size
* Major expenditure categories (food, housing, education, transportation, etc.)

### Target / Label Column (Dependent Variable)
The target variable is the engineered class label: `Income_Class`. This represents a **Binary Classification** problem.

| Label | Group | Definition |
| :---: | :--- | :--- |
| **0** | **Low-Income Group** | Families with an annual income at or **below** the national median income. |
| **1** | **High-Income Group** | Families with an annual income **above** the national median income. |

### Objective of the Study
The objective is to build a robust **machine learning classification model** (e.g., Random Forest) that accurately predicts whether a Filipino family belongs to the High-Income (1) or Low-Income (0) group based on their observed demographic characteristics and expenditure patterns. The study aims to identify the key features that drive this economic classification, providing insights for targeted poverty or welfare policies.

---
# 2. Data Collection / Loading
The dataset used in this study is the Family Income and Expenditure Survey (FIES), publicly available on Kaggle. It contains socio-economic and spending data of Filipino households, based on surveys originally conducted by the Philippine Statistics Authority (PSA). The data is stored in a CSV file format (Family Income and Expenditure.csv) and is imported using the pandas library for data analysis and manipulation.

In [317]:
# Importing necessary libraries
import numpy as np
import scipy as sp
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, r2_score

df = pd.read_csv('Family Income and Expenditure.csv')

In [318]:
# rows and columns of the dataframe
df.shape

(41544, 60)

In [319]:
# displaying the first few rows of the dataframe 
df.head()

Total Household Income Region  Total Food Expenditure  \
0                  480332    CAR                  117848   
1                  198235    CAR                   67766   
2                   82785    CAR                   61609   
3                  107589    CAR                   78189   
4                  189322    CAR                   94625   

  Main Source of Income  Agricultural Household indicator  \
0         Wage/Salaries                                 0   
1         Wage/Salaries                                 0   
2         Wage/Salaries                                 1   
3         Wage/Salaries                                 0   
4         Wage/Salaries                                 0   

   Bread and Cereals Expenditure  Total Rice Expenditure  Meat Expenditure  \
0                          42140                   38300             24676   
1                          17329                   13008             17434   
2                          34182                   32001              7783   
3                          34030                   28659             10914   
4                          34820                   30167             18391   

   Total Fish and  marine products Expenditure  Fruit Expenditure  ...  \
0                                        16806               3325  ...   
1                                        11073               2035  ...   
2                                         2590               1730  ...   
3                                        10812                690  ...   
4                                        11309               1395  ...   

   Number of Refrigerator/Freezer  Number of Washing Machine  \
0                               1                          1   
1                               0                          1   
2                               0                          0   
3                               0                          0   
4                               1                          0   

   Number of Airconditioner  Number of Car, Jeep, Van  \
0                         0                         0   
1                         0                         0   
2                         0                         0   
3                         0                         0   
4                         0                         0   

   Number of Landline/wireless telephones  Number of Cellular phone  \
0                                       0                         2   
1                                       0                         3   
2                                       0                         0   
3                                       0                         1   
4                                       0                         3   

   Number of Personal Computer  Number of Stove with Oven/Gas Range  \
0                            1                                    0   
1                            1                                    0   
2                            0                                    0   
3                            0                                    0   
4                            0                                    0   

   Number of Motorized Banca  Number of Motorcycle/Tricycle  
0                          0                              1  
1                          0                              2  
2                          0                              0  
3                          0                              0  
4                          0                              1  

[5 rows x 60 columns]

---
# 3. Data Information and Summary Statistics
This section provides an overview of the dataset's structure, data types, and the initial statistical distribution of the variables, which is crucial for identifying necessary preprocessing and cleaning steps

In [320]:
# dataframe information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41544 entries, 0 to 41543
Data columns (total 60 columns):
 #   Column                                         Non-Null Count  Dtype 
---  ------                                         --------------  ----- 
 0   Total Household Income                         41544 non-null  int64 
 1   Region                                         41544 non-null  object
 2   Total Food Expenditure                         41544 non-null  int64 
 3   Main Source of Income                          41544 non-null  object
 4   Agricultural Household indicator               41544 non-null  int64 
 5   Bread and Cereals Expenditure                  41544 non-null  int64 
 6   Total Rice Expenditure                         41544 non-null  int64 
 7   Meat Expenditure                               41544 non-null  int64 
 8   Total Fish and  marine products Expenditure    41544 non-null  int64 
 9   Fruit Expenditure                              41544 non-null

In [321]:
df.shape

(41544, 60)

In [322]:
# check dataframe shape if have null values
df.isnull().sum()

Total Household Income                              0
Region                                              0
Total Food Expenditure                              0
Main Source of Income                               0
Agricultural Household indicator                    0
Bread and Cereals Expenditure                       0
Total Rice Expenditure                              0
Meat Expenditure                                    0
Total Fish and  marine products Expenditure         0
Fruit Expenditure                                   0
Vegetables Expenditure                              0
Restaurant and hotels Expenditure                   0
Alcoholic Beverages Expenditure                     0
Tobacco Expenditure                                 0
Clothing, Footwear and Other Wear Expenditure       0
Housing and water Expenditure                       0
Imputed House Rental Value                          0
Medical Care Expenditure                            0
Transportation Expenditure  

In [323]:
# check if household head occupation have null values
df['Household Head Occupation'].isnull().sum()

np.int64(7536)

In [324]:
df.describe()

Total Household Income  Total Food Expenditure  \
count            4.154400e+04            41544.000000   
mean             2.475556e+05            85099.158194   
std              2.868805e+05            51637.947682   
min              1.128500e+04             2947.000000   
25%              1.048950e+05            51017.250000   
50%              1.640795e+05            72985.500000   
75%              2.911385e+05           105635.500000   
max              1.181599e+07           827565.000000   

       Agricultural Household indicator  Bread and Cereals Expenditure  \
count                      41544.000000                   41544.000000   
mean                           0.429858                   25133.723642   
std                            0.676666                   13466.482984   
min                            0.000000                       0.000000   
25%                            0.000000                   16556.000000   
50%                            0.000000                   23324.000000   
75%                            1.000000                   31439.000000   
max                            2.000000                  765864.000000   

       Total Rice Expenditure  Meat Expenditure  \
count            41544.000000      41544.000000   
mean             18195.825005      10540.072574   
std              11727.224350      10497.727757   
min                  0.000000          0.000000   
25%              11020.000000       3353.750000   
50%              16620.000000       7332.500000   
75%              23920.000000      14292.000000   
max             758326.000000     261566.000000   

       Total Fish and  marine products Expenditure  Fruit Expenditure  \
count                                 41544.000000       41544.000000   
mean                                  10529.318337        2550.474990   
std                                    7812.201613        2969.752879   
min                                       0.000000           0.000000   
25%                                    5504.000000        1024.750000   
50%                                    8695.000000        1820.000000   
75%                                   13388.000000        3100.000000   
max                                  188208.000000      273769.000000   

       Vegetables Expenditure  Restaurant and hotels Expenditure  ...  \
count            41544.000000                       41544.000000  ...   
mean              5006.918568                       15436.566075  ...   
std               3300.175236                       23705.195929  ...   
min                  0.000000                           0.000000  ...   
25%               2873.000000                        1930.000000  ...   
50%               4314.000000                        7314.000000  ...   
75%               6304.250000                       19921.250000  ...   
max              74800.000000                      725296.000000  ...   

       Number of Refrigerator/Freezer  Number of Washing Machine  \
count                    41544.000000               41544.000000   
mean                         0.394184                   0.319806   
std                          0.537025                   0.478482   
min                          0.000000                   0.000000   
25%                          0.000000                   0.000000   
50%                          0.000000                   0.000000   
75%                          1.000000                   1.000000   
max                          5.000000                   3.000000   

       Number of Airconditioner  Number of Car, Jeep, Van  \
count              41544.000000              41544.000000   
mean                   0.129814                  0.081215   
std                    0.448673                  0.346786   
min                    0.000000                  0.000000   
25%                    0.000000                  0.000000   
50%                    0.000000                  0.000000   
75%       

## Findings:
- The dataset contains 41,544 rows (representing individual families/households) and 60 columns (features). 
- A check for null values across the entire dataset revealed zero missing entries (as confirmed by the count row being uniform across all columns) except for Household Head Occupation that has 7,536 missing values.
- the data is complete but exhibits high skewness and large differences in feature scales. The next steps in data cleaning and pre-processing must focus on handling the income skew and scaling all numerical features.

---
# 4. Data Cleaning
This section addresses inconsistencies and prepares the raw features for the subsequent Data Engineering steps.

### 4.1 Dropping Unneeded Columns
This process involves permanently removing columns from the DataFrame that are redundant, irrelevant, or potentially harmful to the modeling process.

In [325]:
# list all the columns in the dataframe
pd.DataFrame({'Column Names': df.columns})

Column Names
0                          Total Household Income
1                                          Region
2                          Total Food Expenditure
3                           Main Source of Income
4                Agricultural Household indicator
5                   Bread and Cereals Expenditure
6                          Total Rice Expenditure
7                                Meat Expenditure
8     Total Fish and  marine products Expenditure
9                               Fruit Expenditure
10                         Vegetables Expenditure
11              Restaurant and hotels Expenditure
12                Alcoholic Beverages Expenditure
13                            Tobacco Expenditure
14  Clothing, Footwear and Other Wear Expenditure
15                  Housing and water Expenditure
16                     Imputed House Rental Value
17                       Medical Care Expenditure
18                     Transportation Expenditure
19                      Communication Expenditure
20                          Education Expenditure
21   Miscellaneous Goods and Services Expenditure
22                  Special Occasions Expenditure
23            Crop Farming and Gardening expenses
24   Total Income from Entrepreneurial Acitivites
25                             Household Head Sex
26                             Household Head Age
27                  Household Head Marital Status
28         Household Head Highest Grade Completed
29       Household Head Job or Business Indicator
30                      Household Head Occupation
31                 Household Head Class of Worker
32                              Type of Household
33                 Total Number of Family members
34          Members with age less than 5 year old
35              Members with age 5 - 17 years old
36        Total number of family members employed
37                         Type of Building/House
38                                   Type of Roof
39                                  Type of Walls
40                               House Floor Area
41                                      House Age
42                             Number of bedrooms
43                                  Tenure Status
44                              Toilet Facilities
45                                    Electricity
46                    Main Source of Water Supply
47                           Number of Television
48                           Number of CD/VCD/DVD
49                 Number of Component/Stereo set
50                 Number of Refrigerator/Freezer
51                      Number of Washing Machine
52                       Number of Airconditioner
53                       Number of Car, Jeep, Van
54         Number of Landline/wireless telephones
55                       Number of Cellular phone
56                    Number of Personal Computer
57            Number of Stove with Oven/Gas Range
58                      Number of Motorized Banca
59                  Number of Motorcycle/Tricycle

In [326]:
df.drop(columns=[
    'Total Household Income',  # Must be dropped to ensure the model learns from the predictors and not the answer itself.

    # The informations is sufficiently captured by Total Food Expenditure. Keeping all detailed food items introduces multicollinearity.
    'Bread and Cereals Expenditure',
    'Total Rice Expenditure',
    'Meat Expenditure',
    'Total Fish and  marine products Expenditure', 
    'Fruit Expenditure',
    'Vegetables Expenditure',

    # Redundant: Less Predictive Detailed Expenditures
    'Restaurant and hotels Expenditure', 

    # Redundant: This is a calculated value highly correlated with Housing and water Expenditure and House Floor Area.
    'Imputed House Rental Value', 
    
    # Redundant: with Occupation/Class of Worker
    'Household Head Job or Business Indicator',
    
    # Redundant: with Main Source of Income
    'Total Income from Entrepreneurial Acitivites', 
    
    # Redundant: Information is sufficiently captured by Total Number of Family members.
    'Members with age less than 5 year old',
    'Members with age 5 - 17 years old', 
    
    # Redundant/Correlated: Often correlated with Type of Building/House and House Floor Area.
    'Type of Roof', 
    'Type of Walls', 
    'Toilet Facilities', 
    
    # Low Variance: These are standard utilities for most households; they offer little power to differentiate between income classes.
    'Electricity', 
    'Main Source of Water Supply',

    # Low Differential/Obsolete: Less discriminatory for high income than a car or AC unit. CD/VCD are obsolete indicators.
    'Number of Television', 
    'Number of CD/VCD/DVD', 
    'Number of Component/Stereo set', 
    
    #Obsolete Indicator: Mobile phones are the standard.
    'Number of Landline/wireless telephones',

    #Noise/Redundant: Already covered by Main Source of Income.
    'Agricultural Household indicator',

    # Noise: Highly variable and not a consistent economic class indicator.
    'Alcoholic Beverages Expenditure',
    'Tobacco Expenditure',

    # Noise/Less Predictive: Less consistent predictor than housing/education. Households may not necessarily spend more if they earn a lot.
    'Clothing, Footwear and Other Wear Expenditure',

    # Noise: Too broad and captures inconsistent spending.
    'Miscellaneous Goods and Services Expenditure',

    # Noise: Random, non-structural spending that does not reflect structural income.
    'Special Occasions Expenditure',

    # Noise/Redundant: Covered by Main Source of Income.
    'Crop Farming and Gardening expenses',
    
    # Less Variance/Redundant: Covered by Marital Status and Family Size.
    'Type of Household',
    
    # Noisy/Unreliable: Often difficult to measure reliably and less predictive than Floor Area.
    'House Age',

    # Niche/Less Differential: Niche vehicles; Car/Jeep/Van is the superior differentiator.
    'Number of Motorized Banca', 
    'Number of Motorcycle/Tricycle'
], inplace=True)

In [327]:
# list all the columns in the dataframe after dropping unneeded columns
pd.DataFrame({
    'Column': df.columns,
    'Type': df.dtypes.values
})

Column    Type
0                                    Region  object
1                    Total Food Expenditure   int64
2                     Main Source of Income  object
3             Housing and water Expenditure   int64
4                  Medical Care Expenditure   int64
5                Transportation Expenditure   int64
6                 Communication Expenditure   int64
7                     Education Expenditure   int64
8                        Household Head Sex  object
9                        Household Head Age   int64
10            Household Head Marital Status  object
11   Household Head Highest Grade Completed  object
12                Household Head Occupation  object
13           Household Head Class of Worker  object
14           Total Number of Family members   int64
15  Total number of family members employed   int64
16                   Type of Building/House  object
17                         House Floor Area   int64
18                       Number of bedrooms   int64
19                            Tenure Status  object
20           Number of Refrigerator/Freezer   int64
21                Number of Washing Machine   int64
22                 Number of Airconditioner   int64
23                 Number of Car, Jeep, Van   int64
24                 Number of Cellular phone   int64
25              Number of Personal Computer   int64
26      Number of Stove with Oven/Gas Range   int64

## Summary:
- Dropped the original Total Household Income column and 33 noisy/redundant/detailed columns to ensure the model learns from the predictors and not the answer itself and prevents multicollinearity.
- All rows have inputs and is not null.

---
# 5. Data Engineering / pre-processing
This section transform the cleaned, consistent data into a format that machine learning algorithms can efficiently process. It is primarily focused on making the features mathematically and statistically suitable for modeling.

### 5.1 Feature Encoding (Handling Categorical Data)
Machine learning models only understand numbers, so any text-based (categorical) data must be converted into numerical representations.

In [328]:
# check all object type columns
df.dtypes[df.dtypes == 'object']

Region                                    object
Main Source of Income                     object
Household Head Sex                        object
Household Head Marital Status             object
Household Head Highest Grade Completed    object
Household Head Occupation                 object
Household Head Class of Worker            object
Type of Building/House                    object
Tenure Status                             object
dtype: object

In [329]:
# check first 5 rows of object type columns
df.select_dtypes(include=['object']).head()

Region Main Source of Income Household Head Sex  \
0    CAR         Wage/Salaries             Female   
1    CAR         Wage/Salaries               Male   
2    CAR         Wage/Salaries               Male   
3    CAR         Wage/Salaries               Male   
4    CAR         Wage/Salaries               Male   

  Household Head Marital Status  \
0                        Single   
1                       Married   
2                       Married   
3                       Married   
4                       Married   

             Household Head Highest Grade Completed  \
0  Teacher Training and Education Sciences Programs   
1                       Transport Services Programs   
2                                           Grade 3   
3                               Elementary Graduate   
4                               Elementary Graduate   

                           Household Head Occupation  \
0  General elementary education teaching professi...   
1                               Transport conductors   
2                             Farmhands and laborers   
3                                       Rice farmers   
4  General managers/managing proprietors in trans...   

                     Household Head Class of Worker Type of Building/House  \
0      Worked for government/government corporation           Single house   
1                  Worked for private establishment           Single house   
2                  Worked for private establishment           Single house   
3  Employer in own family-operated farm or business           Single house   
4                 Self-employed wihout any employee           Single house   

                                   Tenure Status  
0  Own or owner-like possession of house and lot  
1  Own or owner-like possession of house and lot  
2  Own or owner-like possession of house and lot  
3  Own or owner-like possession of house and lot  
4  Own or owner-like possession of house and lot

In [330]:
# For specific columns
columns_to_check = ['Region', 'Main Source of Income','Household Head Sex', 
                    'Household Head Marital Status', 'Household Head Highest Grade Completed',
                    'Household Head Occupation', 'Household Head Class of Worker',
                    'Type of Building/House', 'Tenure Status']

for col in columns_to_check:
    print(f"\n{col}:")
    print(df[col].unique())


Region:
['CAR' 'Caraga' 'VI - Western Visayas' 'V - Bicol Region' ' ARMM'
 'III - Central Luzon' 'II - Cagayan Valley' 'IVA - CALABARZON'
 'VII - Central Visayas' 'X - Northern Mindanao' 'XI - Davao Region'
 'VIII - Eastern Visayas' 'I - Ilocos Region' 'NCR' 'IVB - MIMAROPA'
 'XII - SOCCSKSARGEN' 'IX - Zasmboanga Peninsula']

Main Source of Income:
['Wage/Salaries' 'Other sources of Income' 'Enterpreneurial Activities']

Household Head Sex:
['Female' 'Male']

Household Head Marital Status:
['Single' 'Married' 'Widowed' 'Divorced/Separated' 'Annulled' 'Unknown']

Household Head Highest Grade Completed:
['Teacher Training and Education Sciences Programs'
 'Transport Services Programs' 'Grade 3' 'Elementary Graduate'
 'Second Year High School' 'Third Year High School'
 'Business and Administration Programs' 'First Year College'
 'High School Graduate'
 'Other Programs in Education at the Third Level, First Stage, of the Type that Leads to an Award not Equivalent to a First University or 

In [331]:
# print the count of unique values in each column
df_to_check_without_int_col = df.select_dtypes(exclude=['int64', 'float64'])

for col in columns_to_check:
    unique_counts = df_to_check_without_int_col.nunique()

print("\nUnique value counts for categorical columns:\n")
print(unique_counts)


Unique value counts for categorical columns:

Region                                     17
Main Source of Income                       3
Household Head Sex                          2
Household Head Marital Status               6
Household Head Highest Grade Completed     46
Household Head Occupation                 378
Household Head Class of Worker              7
Type of Building/House                      6
Tenure Status                               8
dtype: int64


### Findings:
- columns that has low cardinality (Region, Main Source of Income, etc.) can be encode using **one-hot encoding** and **binary-encoding** (Male = 0, Female = 1)
- columns with too many cardinality like **Household Head Highest Grade Completed** and **Household Head Occupation** needs to be encoded in different way

---

### 5.1.2 One-hot Encoding and Binary Encoding Columns
To prepare the dataset for machine learning, categorical variables must be converted into numerical form so that the Random Forest Classifier can interpret them effectively. Two encoding techniques were used in this study — **One-Hot Encoding and Binary Encoding** — depending on the number of unique categories in each column.

- **One-hot encoding** - was applied to categorical features with a small number of unique values.
- **Binary encoding** - was used for features containing only two categories.


In [332]:
print(f"Original shape: {df.shape}")

# List of columns to one-hot encode
categorical_columns = [
    'Region',
    'Main Source of Income',
    'Household Head Marital Status',
    'Household Head Class of Worker',
    'Type of Building/House',
    'Tenure Status'
]

# One-hot encode the categorical columns
df = pd.get_dummies(df, columns=categorical_columns, drop_first=False)

from sklearn.preprocessing import LabelEncoder

# Encode Sex column (Female = 0, Male = 1)
le = LabelEncoder()
df['Household Head Sex'] = le.fit_transform(df['Household Head Sex'])


# Display the shape before and after encoding

print(f"Encoded shape: {df.shape}")



Original shape: (41544, 27)
Encoded shape: (41544, 68)


---
### 5.1.3 Ordinal Encoding - Household Head Highest Grade Completed (46 Categories)
The “Household Head Highest Grade Completed” column contains 46 unique categories representing various levels of educational attainment. Since education follows a natural progression or order (from no education to postgraduate studies), it is most appropriate to represent this feature using Ordinal Encoding.

Ordinal Encoding converts each category into a numerical value based on its relative rank or educational level. This approach preserves the inherent order among categories, allowing the model to interpret higher numerical values as higher levels of education.

In [333]:
# All specific 'Programs' are assigned the same high rank (20) as they imply a degree/specialization.
education_rank_map = {
    # ------------------ Rank 1-9: Elementary School ------------------
    'No Grade Completed': 1,
    'Basic Programs': 2,
    'Preschool': 2,
    'Grade 1': 3,
    'Grade 2': 4,
    'Grade 3': 5,
    'Grade 4': 6,
    'Grade 5': 7,
    'Grade 6': 8,
    'Elementary Graduate': 9,

    # ------------------ Rank 10-13: High School ------------------
    'First Year High School': 10,
    'Second Year High School': 11,
    'Third Year High School': 12,
    'High School Graduate': 13,

    # ------------------ Rank 14-15: Post Secondary ------------------
    'First Year Post Secondary': 14,
    'Second Year Post Secondary': 15,

    # ------------------ Rank 16-19: College Years ------------------
    'First Year College': 16,
    'Second Year College': 17,
    'Third Year College': 18,
    'Fourth Year College': 19,

    # ------------------ Rank 20: Baccalaureate/Degree Programs ------------------
    # All 25 Program categories are ranked equally at 20 (Degree Equivalent)
    'Teacher Training and Education Sciences Programs': 20,
    'Transport Services Programs': 20,
    'Business and Administration Programs': 20,
    'Other Programs in Education at the Third Level, First Stage, of the Type that Leads to an Award not Equivalent to a First University or Baccalaureate Degree': 20,
    'Humanities Programs': 20,
    'Engineering and Engineering Trades Programs': 20,
    'Engineering and Engineering trades Programs': 20,
    'Social and Behavioral Science Programs': 20,
    'Agriculture, Forestry, and Fishery Programs': 20,
    'Health Programs': 20,
    'Security Services Programs': 20,
    'Computing/Information Technology Programs': 20,
    'Mathematics and Statistics Programs': 20,
    'Personal Services Programs': 20,
    'Law Programs': 20,
    'Journalism and Information Programs': 20,
    'Architecture and Building Programs': 20,
    'Manufacturing and Processing Programs': 20,
    'Life Sciences Programs': 20,
    'Other Programs of Education at the Third Level, First Stage, of the Type that Leads to a Baccalaureate or First University/Professional Degree (HIgher Education Level, First Stage, or Collegiate Education Level)': 20,
    'Social Services Programs': 20,
    'Physical Sciences Programs': 20,
    'Arts Programs': 20,
    'Veterinary Programs': 20,
    'Environmental Protection Programs': 20,

    # ------------------ Rank 21: Post Graduate ------------------
    'Post Baccalaureate': 21
}

column_name = 'Household Head Highest Grade Completed'

df['Education_Rank'] = df[column_name].map(education_rank_map)

df['Education_Rank'] = df['Education_Rank'].fillna(0).astype(int)

df = df.drop(columns=[column_name])

print("Ordinal Encoding applied successfully.\n")
print(df[['Education_Rank']].head())

Ordinal Encoding applied successfully.

   Education_Rank
0              20
1              20
2               5
3               9
4               9


---
### 5.1.4 Frequency Encoding - Household Head Occupation (378 Categories)
The “Household Head Occupation” column contains 378 unique categories, representing a wide range of job titles and professions. Because of its high cardinality, using one-hot encoding would result in an extremely large number of columns, leading to increased computational cost and potential model overfitting. To address this, the column was transformed using **Frequency Encoding**.

**Frequency Encoding** - replaces each category with the number of times it appears in the dataset or its relative frequency (i.e., proportion of total records). This method effectively converts the categorical variable into a numerical feature that reflects how common or rare each occupation is.

In [334]:
occupation_col = 'Household Head Occupation'
freq_col = f'{occupation_col}_freq'

df[occupation_col] = df[occupation_col].fillna('Unknown_Occupation')
freq_encoding = df[occupation_col].value_counts().to_dict()
df[freq_col] = df[occupation_col].map(freq_encoding).astype(float)

# Check missing values after encoding
print(df.isnull().sum())



Total Food Expenditure                                            0
Housing and water Expenditure                                     0
Medical Care Expenditure                                          0
Transportation Expenditure                                        0
Communication Expenditure                                         0
                                                                 ..
Tenure Status_Rent house/room including lot                       0
Tenure Status_Rent-free house and lot with consent of owner       0
Tenure Status_Rent-free house and lot without consent of owner    0
Education_Rank                                                    0
Household Head Occupation_freq                                    0
Length: 69, dtype: int64


---
## 5.2 Create Target Variable
To enable supervised learning, the dataset required a target variable representing household income class. The “Total Household Income” feature was used as the primary indicator for this purpose.

A new binary column, Income_Class, was created to classify households into two groups:

0 – Low Income Class

1 – High Income Class

#### PHILIPPINE INCOME CLASSIFICATION (Based on PSA Standards)
According to the Philippine Statistics Authority (PSA), the official poverty threshold varies by region and family size. However, for a standardized classification, use the following approach:

- Method: Official PSA Poverty Threshold National Average Annual Poverty Threshold per Family: PHP 166,452 (or PHP 13,873 per month) This is the minimum income needed to meet basic food and non-food needs which is close to the median PHP 164,079.50. 

The threshold will be used for the study. Households or individuals earning below or equal to PHP 166,452 are classified as Low Income, while those earning above it are classified as High Income.
This binary categorization allows for a simplified yet meaningful model of income disparity, reflecting real-world socioeconomic divisions.


In [ ]:
# Load the original dataset to access the 'Total Household Income' column
df_original = pd.read_csv('Family Income and Expenditure.csv')
income = df_original['Total Household Income']

PSA_POVERTY_THRESHOLD = 166452.50  # Annual per family

# Low-Income (0): Below poverty threshold
# High-Income (1): Above
#  poverty threshold
df['Income_Class'] = (income >= PSA_POVERTY_THRESHOLD).astype(int)


---
## check niyo na lang gclass ni maam para sa guide
# 6. Data Visualization 
# 7. Data Splitting (Training and Testing Set)
# 8. Model Selection and Training
# 9. Prediction Values (if applicable)
# 10. Visualization of Model Analysis
# 11. Performance Metrics
# 12. Conclusion